sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala

```diff
@@ -168,6 +170,57 @@ class StatisticsSuite extends QueryTest with TestHiveSingleton with SQLTestUtils
       TableIdentifier("tempTable"), ignoreIfNotExists = true, purge = false)
   }
 
+  test("generate table-level statistics") {
+    def checkTableStats(
+        statsSeq: Seq[Statistics],
+        sizeInBytes: Int,
+        estimatedSize: Int,
+        rowCount: Int): Unit = {
+      assert(statsSeq.size === 1)
+      assert(statsSeq.head.sizeInBytes === BigInt(sizeInBytes))
+      assert(statsSeq.head.estimatedSize === Some(BigInt(estimatedSize)))
+      assert(statsSeq.head.rowCount === Some(BigInt(rowCount)))
+    }
+
+    sql("CREATE TABLE analyzeTable (key STRING, value STRING)").collect()
+    sql("CREATE TABLE parquetTable (key STRING, value STRING) STORED AS PARQUET").collect()
+    sql("CREATE TABLE orcTable (key STRING, value STRING) STORED AS ORC").collect()
+
+    sql("INSERT INTO TABLE analyzeTable SELECT * FROM src").collect()
+    sql("INSERT INTO TABLE parquetTable SELECT * FROM src").collect()
+    sql("INSERT INTO TABLE orcTable SELECT * FROM src").collect()
+    sql("INSERT INTO TABLE orcTable SELECT * FROM src").collect()
+
+    sql("ANALYZE TABLE analyzeTable COMPUTE STATISTICS")
```

Comment:
Before calling `ANALYZE TABLE`, table properties already have the latest values of `numFiles` and `totalSize`. Thus, `totalSize` is not added/updated by `ANALYZE TABLE`.